hostpwrctl.cpp

```diff
@@ -0,0 +1,414 @@
+        systemBus.call(method).read(data);
+        return std::move(std::get<std::string>(data));
+    }
+    catch (const sdbusplus::exception::SdBusError& e)
+    {
+        fprintf(stderr, "Error occured during get property request, %s\n",
+                e.what());
+    }
+
+    return "";
+}
+
+/**
+ * @brief Set D-Bus property
+ *
+ * @param path     - object path
+ * @param iface    - D-Bus interface
+ * @param property - property name
+ * @param value    - new value
+ */
+void setProperty(const std::string& path, const std::string& iface,
+                 const std::string& property, const std::string& value)
+{
+    auto service = getService(path, iface);
+    if (service.empty())
+    {
+        return;
+    }
+
+    auto method = systemBus.new_method_call(service.c_str(), path.c_str(),
+                                            ifaceDBusProperties, "Set");
+    std::variant<std::string> data(value);
+    method.append(iface, property, data);
+
+    try
+    {
+        systemBus.call(method);
+    }
+    catch (const sdbusplus::exception::SdBusError& e)
+    {
+        fprintf(stderr, "Error occured during set property request, %s\n",
+                e.what());
+    }
+}
+
+/**
+ * @brief Terminate the event loop if expected values reached
+ */
+void checkExpectations()
+{
+    if (!expectedHostState.empty() && !expectedChassisState.empty() &&
+        expectedHostState == currentHostState &&
+        expectedChassisState == currentChassisState)
+    {
+        systemEvent.exit(EXIT_SUCCESS);
+    }
+}
+
+/**
+ * @brief PropertiesChanged signal handler
+ *
+ * @param m - signal data
+ */
+void onPropertiesChanged(sdbusplus::message::message& m)
+{
+    std::string iface;
+    std::map<std::string, std::variant<std::string>> data;
+    std::vector<std::string> v;
+
+    m.read(iface, data, v);
+
+    if (iface == chassisIface)
+    {
+        auto it = data.find(chassisState);
+        if (it != data.end())
+        {
+            currentChassisState = std::get<std::string>(it->second);
+            printf("Current Chassis State: %s\n",
+                   trimClassName(currentChassisState).c_str());
+            checkExpectations();
+        }
+    }
+    else if (iface == hostIface)
+    {
+        auto it = data.find(hostState);
+        if (it != data.end())
+        {
+            currentHostState = std::get<std::string>(it->second);
+            printf("Current Host State: %s\n",
+                   trimClassName(currentHostState).c_str());
+            checkExpectations();
+        }
+    }
+}
+
+/**
+ * @brief Send the power on command
+ */
+void switchHostPowerOn(sdeventplus::source::EventBase&)
+{
+    if (currentChassisState != chassisStateOn)
+    {
+        expectedHostState = hostStateOn;
+        expectedChassisState = chassisStateOn;
+        setProperty(hostPath, hostIface, hostTransition, hostTransitionOn);
+        printf(
+            "Powerup signal sent to host, waiting to power the system on.\n");
```

Comment:
"The powerup signal was sent to the host", "waiting for".
But I would not use articles for such messages, no one writes "The file was not found".

Reply:
done

Reply:
What is it waiting for? :-))))
"waiting for system start"?

Reply:
Written as you suggested. ;)
fixed.